test/unittest/rtps/resources/timedevent/TimedEventTests.cpp

```diff
@@ -276,188 +161,81 @@ TEST(TimedEvent, EventOnSuccessAutoDestruc_RestartEvents)
+TEST(TimedEvent, Event_AutoRestart)
 {
-    // Restart destriction counter.
-    MockEvent::destructed_ = 0;
-    MockEvent *event = new MockEvent(env->service_, *env->thread_, 10 , true);
+    MockEvent event(env->service_, 10 , true);
 
     for(unsigned int i = 0; i < 100; ++i)
     {
-        event->restart_timer();
+        event.event().restart_timer();
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
     }
 
-    int successed = event->successed_.load(std::memory_order_relaxed);
+    int successed = event.successed_.load(std::memory_order_relaxed);
 
     ASSERT_GE(successed , 100);
-
-    delete event;
-
-    std::unique_lock<std::mutex> lock(MockEvent::destruction_mutex_);
-
-    if(MockEvent::destructed_ != 1)
-    {
-        MockEvent::destruction_cond_.wait_for(lock, std::chrono::milliseconds(100));
-    }
-
-    ASSERT_EQ(MockEvent::destructed_, 1);
 }
 
 
 /*!
- * @fn TEST(TimedEvent, EventNonAutoDestruc_AutoRestartAndDeleteRandomly)
+ * @fn TEST(TimedEvent, Event_AutoRestartAndDeleteRandomly)
  * This test checks an event, configured to restart itself, can be deleted while
  * it is being scheduled.
  * This test launches an event that restarts itself, and then randomly deletes it.
  */
-TEST(TimedEvent, EventNonAutoDestruc_AutoRestartAndDeleteRandomly)
+TEST(TimedEvent, Event_AutoRestartAndDeleteRandomly)
 {
-    // Restart destriction counter.
-    MockEvent::destructed_ = 0;
-
     std::random_device rd;
     std::mt19937 gen(rd());
     std::uniform_int_distribution<> dis(10, 100);
 
-    MockEvent* event = new MockEvent(env->service_, *env->thread_, 2, true);
+    MockEvent event(env->service_, 2, true);
 
-    event->restart_timer();
+    event.event().restart_timer();
     std::this_thread::sleep_for(std::chrono::milliseconds(dis(gen)));
-
-    delete event;
-
-    std::unique_lock<std::mutex> lock(MockEvent::destruction_mutex_);
-
-    if(MockEvent::destructed_ != 1)
-    {
-        MockEvent::destruction_cond_.wait_for(lock, std::chrono::milliseconds(100));
-    }
-
-    ASSERT_EQ(MockEvent::destructed_, 1);
 }
 
 /*!
  * @fn TEST(TimedEvent, ParentEventNonAutoDestruc_InternallyDeleteEventNonAutoDestruct)
  * This test checks an event can delete other event while the later is waiting.
  * This test launches an event that internally will destroy other event.
  */
-TEST(TimedEvent, ParentEventNonAutoDestruc_InternallyDeleteEventNonAutoDestruct)
+// Not permitted in current implementation
+/*TEST(TimedEvent, ParentEvent_InternallyDeleteEventNonAutoDestruct)
 {
-    // Restart destriction counter.
-    MockEvent::destructed_ = 0;
-
-    MockParentEvent event(env->service_, *env->thread_, 10, 2);
-
-    event.restart_timer();
-
-    std::unique_lock<std::mutex> lock(MockEvent::destruction_mutex_);
-
-    if(MockEvent::destructed_ != 1)
-    {
-        MockEvent::destruction_cond_.wait_for(lock, std::chrono::milliseconds(300));
-    }
+    MockParentEvent event(env->service_, 10, 2);
```

Comment:
Should this test be removed if it is no longer permitted in the current implementation?

Reply:
Removed